pkg/trace/writer/sender.go

```diff
@@ -155,14 +157,34 @@ func (s *queuableSender) Start() {
 // Run executes the queuableSender main logic synchronously.
 func (s *queuableSender) Run() {
 	defer close(s.exit)
+	sema := make(chan struct{}, s.conf.MaxConnections)
 
 	for {
 		select {
 		case payload := <-s.in:
-			if stats, err := s.sendOrQueue(payload); err != nil {
-				log.Debugf("Error while sending or queueing payload. err=%v", err)
-				s.notifyError(payload, err, stats)
+			if payload == nil {
+				continue
+			}
+			s.mu.RLock()
+			queuing := s.queuedPayloads.Len() > 0
+			s.mu.RUnlock()
+			if queuing {
```

Comment:
Nit: would be nice to be able to just do `if s.queuing() { ...`  and have the locking done inside there.